framework/platform/bpl/uci/db/bpl_db_uci.cpp

```diff
@@ -44,7 +44,48 @@ bool uci_section_exists(const std::string &package_name, const std::string &sect
     LOG(TRACE) << "uci_section_exists() " << package_name << ".(" << section_type << ")"
                << section_name;
 
-    return true;
+    auto ctx = alloc_context();
+    if (!ctx) {
+        return false;
+    }
+
+    char path[MAX_UCI_BUF_LEN] = {0};
+    if (sprintf(path, package_path, package_name.c_str()) <= 0) {
```

Comment:
use `snprintf` as it's safer

Reply:
or better yet, create helper functions to compose those strings instead of c APIs

Reply:
Changed throughout